Update the statistic info of hash join temporary files

yaowangm opened the pull request:
1. Remove the max/min file size in the statistic info of temporary files, and add the average file size.
2. Include the statistic info of temporary files of the hash table and the outer table.

adam8157 commented:
I would suggest having a subject describing what it does, instead of merging codes.

adam8157 commented:
I updated the subject and message since it's not related to the GUC actually, and it doesn't have that test case change.

yaowangm commented:
@adam8157 We must have the test change, since we changed the output of avg file size and the calculation of compression buffer size, which changed the behavior of disabling compression. Otherwise we will break the pipeline. I have updated the test case, please take another look. Thanks!

yaowangm commented:
@adam8157 As talked, I made a minor change to avoid breaking ABI. Please take another look, thanks!

adam8157 commented on src/include/executor/hashjoin.h:
```diff
@@ -409,7 +409,8 @@ typedef struct HashJoinTableData
-	uint64		workset_max_file_size;
+	uint64		workset_avg_file_size;
+	/* Not used, just to for ABI compatibility */
 	uint64		workset_min_file_size;
```
Just rename it to `workset_abi_reserved`.

yaowangm replied:
Done.

adam8157 reviewed (approved):
LGTM, but please rename the not-used variable too, IDE would ignore the comment and suggest it anyway, renaming it will help others.
